contracts/Strategy.sol

```diff
@@ -17,95 +17,389 @@ import {
+     * value.
+     */
+    function prepareMigration(address _newStrategy) internal override{
+        (uint256 _mToken, ) = MMFarmingPool(mmFarmingPool).userInfo(mmFarmingPoolId, address(this));
+        MMFarmingPool(mmFarmingPool).withdraw(mmFarmingPoolId, _mToken);
+		
+        uint256 _mmVault = IERC20(mmVault).balanceOf(address(this));
+        if (_mmVault > 0){
+            IERC20(mmVault).safeTransfer(_newStrategy, _mmVault);
+        }
+	    
+        uint256 _mm = IERC20(mm).balanceOf(address(this));
+        if (_mm > 0){
+            IERC20(mm).safeTransfer(_newStrategy, _mm);
+        }
+    }
 
-    function prepareMigration(address _newStrategy) internal override {
-        // TODO: Transfer any non-`want` tokens to the new strategy
-        // NOTE: `migrate` will automatically forward all `want` in this strategy to the new one
+    /**
+     * Override this to add all tokens/tokenized positions this contract
+     * manages on a *persistent* basis (e.g. not just for swapping back to
+     * want ephemerally).
+     *
+     * NOTE: Do *not* include `want`, already included in `sweep` below.
+     *
+     * Example:
+     *
+     *    function protectedTokens() internal override view returns (address[] memory) {
+     *      address[] memory protected = new address[](3);
+     *      protected[0] = tokenA;
+     *      protected[1] = tokenB;
+     *      protected[2] = tokenC;
+     *      return protected;
+     *    }
+     */
+    function protectedTokens() internal override view returns (address[] memory){
+        address[] memory protected = new address[](1);
+        protected[0] = mmVault;
+        return protected;
+    }
+	
+    function _convertMTokenToWant(uint256 _shares) internal view returns (uint256){
+        uint256 _mTokenTotal = IERC20(mmVault).totalSupply();
+        uint256 _wantInVault = MMVault(mmVault).balance();
+        return (_wantInVault.mul(_shares)).div(_mTokenTotal);
     }
+	
+    // swap $MM for $WBTC
+    function _disposeOfMM() internal {
+        uint256 _mm = IERC20(mm).balanceOf(address(this));
 
-    // Override this to add all tokens/tokenized positions this contract manages
-    // on a *persistent* basis (e.g. not just for swapping back to want ephemerally)
-    // NOTE: Do *not* include `want`, already included in `sweep` below
-    //
-    // Example:
-    //
-    //    function protectedTokens() internal override view returns (address[] memory) {
-    //      address[] memory protected = new address[](3);
-    //      protected[0] = tokenA;
-    //      protected[1] = tokenB;
-    //      protected[2] = tokenC;
-    //      return protected;
-    //    }
-    function protectedTokens()
-        internal
-        override
-        view
-        returns (address[] memory)
-    {}
+        if (_mm >= minMMToSwap) {
+            address[] memory path = new address[](4);
+            path[0] = mm;
+            path[1] = usdc;
+            path[2] = weth;
+            path[2] = wbtc;
```

Comment:
bug here, 0,1,2,2.
Any chance we can make it a single step? Worst case mm->weth->wbtc
It would be nice you talk to maki and add some liquidity to sushi.